Release 0.1.6B

Fueredoriku opened the pull request:
🎉

ddabble committed:
Improved enemy logic when target is destroyed

ddabble committed:
Moved Enemies folder -> Prefabs/Enemy

ddabble committed:
Moved Skeleton folder -> Scripts/Enemy/Enemies

ddabble committed:
Cleaned up viewing state variables for debugging

...in the Unity inspector.

ddabble committed:
Moved generic code from SkeletonController to Enemy

ddabble committed:
Added vector distance comparison extension methods

Fueredoriku committed:
Made playerspecificinputmanager spawn different playermodels for different controllers. Renamed models.

ddabble committed:
Improved enemy target logic

More checks are now in place, to avoid `NullReferenceException`s, and the cases when the target does no longer exist, are now handled more gracefully.

Also removed the unused `Enemy.Walk()` and `SkeletonEventHelper.OnWalkEvent()` methods, and the unused `SkeletonWalkBehaviour` script.

ddabble committed:
Made EnemyManager spawn skeletons

ddabble committed:
Moved more setup code from Start() to Awake()

Placing code for initializing things like component references, in the `Awake()` method, follows best practice.

Also moved other code related to this setup code, closer to where it's defined, like delegate subscription methods and the `OnDestroy()` method
(which largely contains the opposite logic to that of `Awake()`).

ddabble committed:
Misc. code improvements

ddabble committed:
Fixed EnemyManager exception when restarting game

E.g. after the base has been destroyed, which automatically restarts the game.

ddabble committed:
Replaced copy of prefab with reference to prefab

...in `Dahl_Player.prefab`.

ddabble committed:
Moved SkeletonRagdollController to proper folder

ddabble committed:
Moved loot models and materials to proper folder

ddabble committed:
Renamed old `enemy` model folder -> PlaceholderEnemy

ddabble committed:
Fixed skeletons not dropping loot

ddabble committed:
Fixed loot spawning with wrong rotation

Also tweaked the spawned world position.

SondreHus committed:
Merge pull request #58 from hackerspace-ntnu/feature-skellyboi

Feature skellyboi

ddabble committed:
Set player state to FREE after canceling UI wheel

Or, in other words, after letting go of the `Select` button without having selected anything.

ddabble committed:
Removed setting player state in InstantiateConstructionTower

...as it's not part of the "task" of that method, which is to simply instantiate the construction tower.

ddabble committed:
UIControllerWheel uses index for selected segment

Also removed storing the same data twice, by replacing the `PlayerUi.selectedSegment` field with getting the selected segment from UIControllerWheel.

ddabble committed:
Fixed wheel segment scaling animation not finishing

...which caused the "scale down" animation to play when reopening the tower construction wheel, after having previously selected a tower to build.

ddabble committed:
Fixed not initially setting construction tower position

...which caused the construction tower to be visible at the world origin for a few frames after selecting it from the radial menu.

ddabble committed:
Fixed not always focusing on held interactable

This was made apparent when looking towards and moving too close to an existing tower, while in building mode, which caused exceptions to be thrown.

ddabble committed:
Renamed Decor folder -> Scenery

ddabble committed:
Moved terrain scenery materials to proper folder

ddabble committed:
Moved lightning tower VFX to proper folder

ddabble committed:
Moved scripts to top of components in prefabs

Scripts are usually the components that are most used/tweaked, and so it's a useful convention to place all scripts at the top of component lists.

ddabble committed:
Moved player-related prefabs to Player folder

Also renamed `Players` folder to `PlayerCharacters`.

ddabble committed:
Moved prefabs from top-level to proper folders

ddabble committed:
Moved ScriptableObject scripts to Scripts folder

ddabble committed:
Renamed TowerScript -> TowerScriptableObject

ddabble committed:
Removed the unused TowerConstructor script

Fueredoriku committed:
manual merge.

ddabble committed:
Fixed placing more than one tower on a single cell

ddabble committed:
Merge pull request #61 from hackerspace-ntnu/fix/misc

Fix misc. bugs + code improvements

ddabble committed:
Merge remote-tracking branch 'origin/develop' into pirate-lady

# Conflicts:
#	Assets/Prefabs/Player/PlayerCharacters/Dahl_Player.prefab
#	Assets/Prefabs/Player/PlayerCharacters/Dahl_Player.prefab.meta
#	Assets/Prefabs/Player/PlayerCharacters/Player_Dahl.prefab
#	Assets/Prefabs/Player/PlayerCharacters/Player_Dahl.prefab.meta
#	Assets/Prefabs/Player/PlayerCharacters/Player_Pirate.prefab
#	Assets/Prefabs/Player/PlayerCharacters/Player_Pirate.prefab.meta
#	Assets/Prefabs/Players/Dahl_Player.prefab
#	Assets/Prefabs/Players/Dahl_Player.prefab.meta

ddabble committed:
Tiny code cleanup

ddabble committed:
Fixed exception for non-existent interactables

ddabble committed:
Cleaned up file placements

...specifically files in the `Animations`, `Materials`, `Models` and `Textures` folders.

ddabble committed:
Cleaned up Player_Pirate prefab component order

ddabble committed:
Replaced copy of prefab with reference to prefab

...in `Player_Pirate.prefab` - same as in dbdd4e4708c082b1c13279203bce3740a6dee1e4.

Fueredoriku committed:
Merge pull request #60 from hackerspace-ntnu/pirate-lady

Pirate lady

imje committed:
Added strongman model

Commit:
Discovered issue where towers could not be placed on terrain. Temporary fix is to rebuild the terrain on Awake. However this resets all tree and decoration positions.

Commit:
Merge remote-tracking branch 'origin/develop' into terrain-update-II

# Conflicts:
#	Assets/Prefabs/Terrain/HexCell.prefab
#	Assets/Prefabs/Terrain/Scenery/New Material 1.mat
#	Assets/Prefabs/Terrain/Scenery/New Material 1.mat.meta
#	Assets/Prefabs/Terrain/Scenery/New Material.mat
#	Assets/Prefabs/Terrain/Scenery/New Material.mat.meta
#	Assets/Prefabs/Terrain/Scenery/Rocks.prefab
#	Assets/Prefabs/Terrain/Scenery/Rocks.prefab.meta
#	Assets/Scenes/Main.unity
#	Assets/Scripts/Enemy/Enemy.cs
#	Assets/Scripts/Enemy/EnemyManager.cs
#	Assets/Scripts/GameState/BaseController.cs
#	Assets/Scripts/Terrain/HexCell.cs
#	Assets/Scripts/Terrain/HexGrid.cs
#	Assets/Scripts/Terrain/HexMapEditor.cs

Fueredoriku committed:
Updated terrain to use prefabs transform.rotation instead of their own. Updated decor-related prefabs to have correct rotation.

Fueredoriku committed:
Removed collider from grass decor.

Fueredoriku committed:
Added hurtBox script and hurtbox to scene to make players die when entering water tiles.

Commit:
More improvements to terrain!

Removed unnecessary terrain code
Terrain no longer rebuilds every time the scene loads, allowing for custom maps
Created a save system which uses binary encoding to store map information
Terrain can currently save and load decorations, cell elevations, turrets and their rotations.
Cleaned up the tree prefab decoration
Edited the HexCell to work easier with decorations and other objects which are children to them.
Created simple UI to save and load a map.

Known issues:
Elevation data is not always correctly rebuilt during an initialization.
Turrets that have been rotated and saved are loaded back in rotated on a wrong axis.

ThomasJGabrielsen committed:
Fixed Terrain initialization bug

ThomasJGabrielsen committed:
Added a RebuildTerrain button

ThomasJGabrielsen committed:
Scene update

imje committed:
New Strongman model and textures!

note: file has light

ThomasJGabrielsen committed:
Small bugfix commit

SaveSystem now closes the stream.
NavMesh is updated, however crashes after a couple seconds.
Cleaned up redundant commented code

ThomasJGabrielsen committed:
Implemening Strongman

todo:
- Rename Animations
- remake certain Animations to make them loopable and blendable
- Create missing assets (blueprint)

Fueredoriku committed:
Updated Strongman prefab to carry items correctly, and move with right rotation. Updated playerinputmanager to spawn strongman as third character.

jesper2k committed:
updated base prefab

jesper2k committed:
restructured base object hierarchy

Fueredoriku committed:
Added a very basic audiomanager, with the song light cavalry overture by von Suppe.

Fueredoriku committed:
Updated base animation to match new hierarchy in prefab.

Fueredoriku committed:
Added AudioManager to Scene.

ThomasJGabrielsen committed:
Merge pull request #64 from hackerspace-ntnu/podium-base

Podium base

ThomasJGabrielsen committed:
Final Fix

Fixed bug where skeletons would die before reaching the base.
Allowed for larger terrains

Fueredoriku committed:
Merge branch 'music-endlessMode' into develop

ThomasJGabrielsen committed:
Merge conflict resolution

ThomasJGabrielsen committed:
Merge branch 'develop' into terrain-update-II

# Conflicts:
#	Assets/Prefabs/Base/Base.prefab
#	Assets/Scenes/Main.unity

magsve committed:
Various decor/models; trees (x2) and a modular-esque stone pile that can be picked apart and put together in different formations for variation.

Fueredoriku committed:
Made terrain a prefab, fixed dependencies in terrain that was lost in merge.

Fueredoriku committed:
Unpacked terrain prefab, as terrain can't be instantiated from prefab (ask Thomas).

ThomasJGabrielsen committed:
Merge pull request #56 from hackerspace-ntnu/terrain-update-II

Terrain update ii

Fueredoriku committed:
Merge pull request #65 from hackerspace-ntnu/decor

Various decor/models; trees (x2) and a modular-esque stone pile that …

ThomasJGabrielsen committed:
Merge remote-tracking branch 'origin/develop' into music-endlessMode

ThomasJGabrielsen committed:
Audio manager added to main scene

ThomasJGabrielsen committed:
Merge pull request #63 from hackerspace-ntnu/music-endlessMode

Basic audio manager + background song for endless mode

ThomasJGabrielsen committed:
Merge remote-tracking branch 'origin/develop' into strongman

ddabble committed:
Removed various old and unused files

More specifically:
* `.terrainlayer` and `.orig` files
* The unused `AudioFIles` folder
* Some old, unused materials and models
* Old and unused scenes (and associated folders - containing things like lightning data and navmeshes)
* The unused `HexMesh` class that had been completely commented out

ddabble committed:
Moved some files to more appropriate folders

* Skeleton, Player and Tower animation files
* Soundtrack files
* Various materials
* Various models
* Palettes
* Base, loot, player and terrain scenery prefabs
* Some scripts
  * Audio scripts
  * Editor scripts
  * `HurtBox` from `Util` to `Universal scripts`
* Base and loot shaders
* PlaceholderEnemy and terrain (scenery) textures
* Various VFX files

ThomasJGabrielsen committed:
EndlessMode

Creates increasing amounts of enemies every x seconds!

ThomasJGabrielsen committed:
Attempt fix of ocean terrain

ThomasJGabrielsen committed:
Merge branch 'fix/terrain-fix' into endless-mode

Fueredoriku committed:
Added basic functionality for repairmechanics and states.

SondreHus committed:
Added speedy golemn and testing scene

SondreHus committed:
Some minor unity automated stuff

SondreHus committed:
Added distortion when base is hit

Fueredoriku committed:
Added basic state-machine structure to handle repair mechanic.

Fueredoriku committed:
Restructured code, update repair function to invoke update for delegate.

Fueredoriku committed:
added repair logic to towers, fixed wron time value bein used in repair script

Fueredoriku committed:
Added animation controller and new vfx files for the repair mechanic.

Fueredoriku committed:
Made basic spark FX.

Fueredoriku committed:
Made basic smoke FX.

ThomasJGabrielsen committed:
More adjustments to terrain.

HexCell no longer has an array of all possible cell types, this is set from hexGrid instead to save memory.

Cells no longer cast shadows by default, instead it's turned on for the highest elevating tiles. This dramatically increases performance as we remove several thousand realtime shadow calculations.

Fueredoriku committed:
Added basic fire VFX.

ddabble committed:
Added an extra blank line after `using` lists

ddabble committed:
Code cleanup

* [HexCell]
  * Removed the unused `position`, `newPos` and `pertubValue` fields and the `Position` property
  * Renamed the serialized `mr` field to `meshRenderer`
* [HexCoordinates] Renamed the serialized `x` and `z` fields to `_x` and `_z`, respectively
* [HexGrid] Renamed the serialized `decor` field to `sceneryObjects`

ddabble committed:
Fixed disabled HexGrid when opening scene

ddabble committed:
Added TextLabel read-only inspector attribute

ddabble committed:
Improved HexCoordinatesDrawer drawing

Now it also draws the field's name as the attribute label in the inspector.

ddabble committed:
Updated README.md with new project name

Also corrected typos and cleaned up the markdown code.

ddabble committed:
Updated gitignore

Fueredoriku committed:
Placed files in a better place in hierarchy.

Fueredoriku committed:
Merge pull request #67 from hackerspace-ntnu/feature/base-hit-distort

Added audio distortion when base is hit

ThomasJGabrielsen committed:
Played around with making edgecells a read only variable, it did not work...

ThomasJGabrielsen committed:
Merge remote-tracking branch 'origin/develop' into fix/terrain_fix_2

# Conflicts:
#	Assets/Prefabs/Terrain/Decor.meta

ddabble committed:
Replaced tab with 4 spaces

ThomasJGabrielsen committed:
Minor adjustment commit;

Moved the .fbx file from Animations to Models
Adjusted for correct animation frames
Changed the carried object transform to Strongman's right hand for extra strongness factor.

Fueredoriku committed:
Added wrappers for each repair-related VFX. Updated RepairAnimationController to instantiate particles based on state. The script should subscribe to RepairController delegate, but that is WIP.

ThomasJGabrielsen committed:
Merge pull request #62 from hackerspace-ntnu/strongman

Strongman

ThomasJGabrielsen committed:
Merge pull request #69 from hackerspace-ntnu/fix/terrain_fix_2

Fix/terrain fix 2

ThomasJGabrielsen committed:
Merge branch 'develop' into endless-mode

ThomasJGabrielsen committed:
Highscore implementation

Added more simple ui to show score
Added a score value to Enemy.cs, so that each enemy prefab gets a custom assigned score.
Scores are added when the enemy dies and Die() is called.

Fueredoriku committed:
Made repair related VFX actually look ok. Made RepairAnimationController utilize the VFX, and updated prefabs to recieve wear over time.

Fueredoriku committed:
Removed old component from prefab.

Fueredoriku committed:
Made interact on tower logic call the repair function. Updated prefabs to recieve damage at a more frequent interval.

Fueredoriku committed:
Merge branch 'develop' into feature/turret-repair

ThomasJGabrielsen committed:
Warning: This commit does not work

Issue lies with loading Highscores.

Changelog:
New menu with new buttons and new Leaderboard!
Minor edits to the animation of the base.
Updated disclaimer text for next release.
New end game screen with buttons and checks and everything

ThomasJGabrielsen committed:
Highscore update!

Added save data checks to check for valid Leaderboard data.
Adjusted GameOverScreen so that it scales with variable resolutions.
Added a bunch of fun default scores based on us developers.

Fueredoriku committed:
Added disclaimer

The out-commented code should be implemented again later when we have a controller/manager that differentiates between endless mode and "normal"-mode.

ThomasJGabrielsen committed:
Scene management now uses names instead of indexes for more readability.

Renamed Scenes to reflect their true purpose.

ThomasJGabrielsen committed:
Merge remote-tracking branch 'origin/endless-mode' into endless-mode

ThomasJGabrielsen committed:
Reset base HP to standard

ThomasJGabrielsen committed:
Merge pull request #71 from hackerspace-ntnu/endless-mode

Endless mode

Fueredoriku committed:
Merge pull request #70 from hackerspace-ntnu/feature/turret-repair

Tower damage and repair

ThomasJGabrielsen committed:
Fixed Leaderboard update bug

Created a custom comparator and restructured how highscores are saved so they can be easily sorted.

ThomasJGabrielsen committed:
Music is now paused when we pause the game.

Added ability to pause individual sources as well as lower the general volume of the AudioListener.

ThomasJGabrielsen committed:
Merge remote-tracking branch 'origin/develop' into fix/misc

# Conflicts:
#	Assets/Animations/Player/Characters/Strongman.meta
#	Assets/Animations/Player/Characters/Strongman/Animations.meta
#	Assets/Animations/Player/Characters/Strongman/Strongman.controller
#	Assets/Animations/Player/Characters/Strongman/Strongman.controller.meta
#	Assets/Palettes/HexGridDrawer.cs
#	Assets/Palettes/HexGridDrawer.cs.meta
#	Assets/Prefabs/Player/Characters/Player_Strongman.prefab
#	Assets/Prefabs/Player/Characters/Player_Strongman.prefab.meta
#	Assets/Scenes/Main/Post processing Profile.asset
#	Assets/Scenes/Terrain.unity
#	Assets/Scripts/GameState/BaseController.cs
#	Assets/Scripts/GameState/Disclaimer.cs
#	Assets/Scripts/GameState/GameOverScreen.cs
#	Assets/Scripts/Terrain/HexCell.cs
#	Assets/Scripts/Terrain/HexGrid.cs
#	Assets/Scripts/UI/ResourceUI.cs

ThomasJGabrielsen committed:
Fixed merge conflicts

Because i have no idea how kdiff works...
Removed Terrain.unity scene, It's no longer in use

ddabble committed:
Moved some tower prefabs to the Prefabs folder

ddabble committed:
Code cleanup

* Made the `Terrain` gameobject in `Endless Mode.unity` into a prefab named `HexGrid.prefab`
* [BaseController] Changed the type of `gameManager` from `GameObject` to `EndlessMode`
* [EndlessMode]
  * Renamed `enemies` -> `enemyPrefabs`, `a` -> `linearSpawnRate` and `k` - > `exponentialSpawnRate`
* [HexCell] Fixed some faulty merging (in b824f56192d2e281fbbd1e9f889d66d42e3e19c3):
  * Re-removed the unused `Position` property
  * Re-renamed the serialized `mr` field to `meshRenderer`
* [HexGrid]
  * Renamed `towers` -> `towerPrefabs`
  * Removed the `SerializeField` attribute from `playerBase`, as it's always set in `Awake()` anyway
* [RepairAnimationController] Renamed the VFX prefab fields (like `fxLow` -> `lowWearVFXPrefab`)
* [RepairController]
  * Renamed `damageInterval` -> `wearInterval` and `timeStamp` -> `lastWearStateChangeTime`
  * Removed the unused `needRepair` field

ddabble committed:
Fixed not passing correct wear state when repairing

ddabble committed:
Made HexGrid a singleton

ddabble committed:
Made HexGrid serialization use a HexCell struct

...instead of three separate arrays (loosely) connected by their shared indices (namely `elevation`, `occupier` and `occupierRotation`
in `TerrainData`).

ThomasJGabrielsen committed:
NavMesh data was missing, baked new.

Fueredoriku committed:
Fixed scene lighting. Removed old terrain navmesh.

Fueredoriku committed:
Merge pull request #68 from hackerspace-ntnu/fix/misc

Clean up large parts of the project

ThomasJGabrielsen committed:
Added a try/catch to check if Highscore data is corrupted or not.

ThomasJGabrielsen committed:
Icomparator was sorting in the wrong order.

Changed "Endie"'s score to 10 so that no score on the leaderboard is 0

magsve committed:
Score count graphics.

ThomasJGabrielsen committed:
Fixed the bug where by one can still pause the game after it has ended.

ddabble committed:
Renamed hex cell materials

Fueredoriku committed:
Made refunding turrets it's own function, and updated the towerScriptableObject with get function. While testing for this bugfix, I found that playerstatecontroller doesn't actually set or update liftedObject, which made the scriptableobject unaccessable. Until this is resolved, the sellTower function in playerstatecontroller will use the holo prefab for cost variable access.

Fueredoriku committed:
Revert "Made refunding turrets it's own function, and updated the towerScriptableObject with get function. While testing for this bugfix, I found that playerstatecontroller doesn't actually set or update liftedObject, which made the scriptableobject unaccessable. Until this is resolved, the sellTower function in playerstatecontroller will use the holo prefab for cost variable access."

This reverts commit 5ff2d9fc57d12f80d7e459eda426c0ec7f63b7df.

Fueredoriku committed:
Redid previous commit after reverting as I managed to push to develop instead of feature branch by mistake.

ddabble committed:
Restored accidentally deleted files

These were accidentally deleted in the 8d3cac1d91ea29188d55badd2e5c22df11c18a44 merge commit.

ddabble committed:
Replaced cell elevation with a HexCellType class

The logic for setting cell elevation now no longer relies on the naively arbitrary indices of the `HexGrid.cellMaterials` list.
It's instead replaced with logic based on an array - `HexGrid.cellTypes` - containing the material and the associated elevation for each type of cell.

Also, moved some of the logic in the previous `HexGrid.ElevateCell()` to the setter of `HexCell.CellType`.

ddabble committed:
Made base centered and aligned with the cell

It's now not offset -0.53 units on the Z-axis anymore, and it's been rotated 30 degrees, so that the player base's hexagonal plinth is aligned with
the hex cell grid.

ddabble committed:
Made HexCell.meshRenderer private

...as there is no need for setting it from elsewhere.

ddabble committed:
Unable to place towers too close to other towers

...and the player base.

ddabble committed:
Renamed tower -> towerPrefab in TurretPrefabConstruction

ddabble committed:
Only call FocusCell() on focus change

This reduces the performance cost.

ddabble committed:
Made turret holos red when unable to build on cell

ddabble committed:
Made turret holo prefabs into prefab variants

The axe, lightning and wheel turret holo prefabs are now prefab variants of `TowerHolo_BasePrefab.prefab`.

Fueredoriku committed:
Reinstated spacing that got lost in translation.

Fueredoriku committed:
Made axe tower instantiate axe shots as children to keep scene more tidy. Reduced lifetime of axe projectiles for gamebalance, as discussed on relevant Trello card.

Fueredoriku committed:
Update Assets/Scripts/Towers/Specific towers/Axe turret/AxeTowerAnimationController.cs

Less bloat = better :)

Co-authored-by: Anders <6058745+ddabble@users.noreply.github.com>

Fueredoriku committed:
Merge pull request #77 from hackerspace-ntnu/fix/axe-tower-children

Axe shot parenting fix

Fueredoriku committed:
Added suggested changes from PR#74.

SondreHus committed:
Added unigolemn material

SondreHus committed:
merged

SondreHus committed:
Minor changes to unigolemn, needs some vfx, but there are more pressing matters

Fueredoriku committed:
Removed superfluous dev comment. Applied Golem material correctly.

Fueredoriku committed:
Merge pull request #78 from hackerspace-ntnu/feature/speedy-boi

Feature/speedy boi

magsve committed:
Lifebar graphics (temporary?) added

jesper2k committed:
Added more beautiful terrain models. Will probably cause a merge conflict.

ddabble committed:
Moved and renamed unigolem files

Fueredoriku committed:
Added HUD prefab with the new UI textures.

ddabble committed:
Small code cleanup

ddabble committed:
Small code cleanup

ddabble committed:
Merge pull request #74 from hackerspace-ntnu/fix/selling-bug

Fixed selling bug

ddabble committed:
Merge remote-tracking branch 'origin/develop' into fix/bugfix

# Conflicts:
#	Assets/Scripts/GameState/GameOverScreen.cs

ddabble committed:
Small code cleanup

ddabble committed:
Merge pull request #72 from hackerspace-ntnu/fix/bugfix

Fix/bugfix

ddabble committed:
Merge remote-tracking branch 'origin/develop' into fix/cleanup-after-merge

# Conflicts:
#	Assets/Scripts/GameState/LeaderboardData.cs
#	Assets/Scripts/UI/Leaderboard.cs

ddabble committed:
Catch additional exception when loading highscores

ddabble committed:
Made "Round Manager" into own prefab

...renamed to `EndlessModeManager`.

Fueredoriku committed:
Added base life bar texture.

ddabble committed:
Fixed not closing highscore stream before deleting

Also improved the code for saving and loading the highscore data, by using `using` statements instead of closing the streams manually.

ddabble committed:
Merge pull request #75 from hackerspace-ntnu/fix/cleanup-after-merge

Cleanup after last merge

ddabble committed:
Merge remote-tracking branch 'origin/develop' into feature/prevent-placing-towers-too-close

# Conflicts:
#	Assets/Prefabs/Towers/AxeTurret/AxeTurretHolo.prefab
#	Assets/Prefabs/Towers/LightningTurret/LightningTurretHolo.prefab
#	Assets/Prefabs/Towers/WheelTurret/WheelTurretHolo.prefab

ddabble committed:
Merge pull request #79 from hackerspace-ntnu/more-terrain-decorations

Added more beautiful terrain models. Will probably cause a merge conf…

ThomasJGabrielsen committed:
Loot rework

Added a drop chance for Loot
Loot now has rigidbodies, so they can be pushed and interacted with physically
Removed obsolote loot variations and cleaned up Loot folder structure
Added the Swedish Palette to an Editor folder, making it accessible for future use.
Revamped the look of the loot crystal, giving it a more similar look to the base crystal.
Adjusted the animation through code to account for the new transparancy effect.

ThomasJGabrielsen committed:
Removed redundant Loot materials

Fixed a bug where the game would crash if an enemy exited the base trigger zone.

ThomasJGabrielsen committed:
Code commenting and adjustment.

ThomasJGabrielsen committed:
Added d-pad support

ThomasJGabrielsen committed:
Switched controller buttons for cancel and interact

SondreHus committed:
Added better post-processing, added light settings, modified renderer, game looks gooder

ddabble committed:
Merge pull request #76 from hackerspace-ntnu/feature/prevent-placing-towers-too-close

Prevent placing towers too close

SondreHus committed:
Merged develop

SondreHus committed:
Fixed multiple deaths occureing

ThomasJGabrielsen committed:
Merge pull request #82 from hackerspace-ntnu/fix/dahl-duplicates

Fixed multiple deaths occurring

magsve committed:
Main menu background, comes in 3 pieces for now!

ThomasJGabrielsen committed:
Players spawn around the base instead of inside it

jesper2k committed:
Added the beautiful ruins prefab to be placed on tall terrain

ddabble committed:
Renamed some unigolem textures

ddabble committed:
Small code cleanup

ddabble committed:
Added some lost changes to main scene after merge

ddabble committed:
Added singleton for EndlessMode

ddabble committed:
Moved accessing singletons to Start()

...as the singleton instances are assigned in the singleton classes' `Awake()` methods.

This also fixes some null pointer exceptions when restarting the game after losing.

ddabble committed:
Merge branch 'develop' into feature/improved-lighting

# Conflicts:
#	Assets/Prefabs/Base/Base.prefab
#	Assets/Prefabs/Towers/TowerHolo_BasePrefab.prefab.meta

ddabble committed:
Merge pull request #81 from hackerspace-ntnu/feature/improved-lighting

Feature/improved lighting

ddabble committed:
Merge remote-tracking branch 'origin/develop' into rework/loot

# Conflicts:
#	Assets/Prefabs/Base/Base.prefab
#	Assets/Scenes/Endless Mode.unity

Fueredoriku committed:
Added moving healthbar to HUD. Updated UImanager to subscribe to base's onDamage delegate.

ThomasJGabrielsen committed:
Added Menu Textures, Trimmed as neccesary

ThomasJGabrielsen committed:
Updated and animated the new main menu

ThomasJGabrielsen committed:
Merge pull request #83 from hackerspace-ntnu/feature/main-menu-background

Main menu background, comes in 3 pieces for now!

Fueredoriku committed:
Merge branch 'develop' into feature/main-menu

ddabble committed:
Small code cleanup

ddabble committed:
Merge pull request #80 from hackerspace-ntnu/rework/loot

Rework/loot

Fueredoriku committed:
Updated position coordinates of ruin prefab to (0,0,0).

Fueredoriku committed:
Merge pull request #84 from hackerspace-ntnu/ruins-prefab

Ruins prefab

ThomasJGabrielsen committed:
Fixed the bug where Highscores with 0 points are highest scores.

It is now not possible to save a score with zero points. If such a score is detected it will reset the Leaderboard.

Fueredoriku committed:
Merge pull request #85 from hackerspace-ntnu/feature/main-menu

HUD

ThomasJGabrielsen committed:
Added logo images and adjusted splash screen

ThomasJGabrielsen committed:
Adjusted ProjectSettings company and name

ThomasJGabrielsen committed:
Switched icon for 1024x1024 version

ddabble committed:
Merge pull request #86 from hackerspace-ntnu/feature/new-main-menu

Feature/new main menu

ddabble committed:
Merge remote-tracking branch 'origin/develop' into rework/controls

ThomasJGabrielsen committed:
Updated game over canvas with new ui elements

ddabble committed:
Code cleanup

* [PlayerStateController.cs] Moved the D-pad code to `PlayerStateController_Input.cs`

ddabble committed:
Set D-pad properties when pressing D-pad buttons

Previously, they were never set.

ddabble committed:
Refactored D-pad action code

ddabble committed:
Small refactor of HealthLogic.DealDamage()

ddabble committed:
Merge pull request #87 from hackerspace-ntnu/rework/controls

Rework/controls

ddabble committed:
Removed unused PlaceholderEnemy files

ddabble committed:
Moved code from Enemy to SkeletonController

This code was in practice just specific to the skeleton enemy, and not enemies in general (like the unigolem).

Also:
* Moved and renamed `Enemy_PlayerDetector` -> `Enemies/Skeleton/Skeleton_PlayerDetector`, as the logic is currently specific to the skeleton enemy,
  and other enemies will not necessarily be interested in trying to detect players
* Renamed `SkeletonController.DealDamage()` -> `OnDealDamage()`
* Renamed `HealthLogic.DealDamage()` -> `OnReceiveDamage()`

ddabble committed:
Made UnigolemController extend Enemy

ddabble committed:
Small code cleanup

ddabble committed:
Added probability weighting to enemy spawning

Also made `EndlessModeManager` spawn unigolems (with a 1/10 probability).

ddabble committed:
Simplified weighted random enemy spawning algorithm

The previous implementation of `ChooseRandomEnemy()` had a time complexity of O(log(n)), but was unnecessarily complex when the list of enemies
will probably always be very short (currently of length 2), and it added a fairly high threshold for understanding and maintaining the code.
This implementation has a time complexity of O(n), but is a lot simpler and should be quicker to understand.

ddabble committed:
Refactored scenery instantiation to HexCell

ddabble committed:
Small refactor of scenery placement methods

ddabble committed:
Added non-occupying scenery generation

Also renamed `HexGrid.sceneryObjects` -> `occupyingSceneryObjects`.

ddabble committed:
Moved scenery prefab objects to ground level

Previously, some of them were placed slightly below or above ground.

ddabble committed:
Resized ruins

They should now be slightly narrower than a hex cell, which looks better when placed on the map.

Fueredoriku committed:
Updated scene direvtly in feature branch with new HUD, as per Thomas request.

Fueredoriku committed:
Merge pull request #88 from hackerspace-ntnu/misc/buid-settings

Misc/buid settings

Fueredoriku committed:
Merge pull request #89 from hackerspace-ntnu/feature/new-main-menu

Updated game over canvas with new ui elements

ThomasJGabrielsen committed:
Merge pull request #92 from hackerspace-ntnu/feature/main-menu

HUD

Fueredoriku committed:
Fixed health variable update from commit bbe59129fab137720b860cab58ac665c0390063d, which immidieatly destroyed base on damage.

ThomasJGabrielsen committed:
Fixed a bug where the player would spawn further and further away from the base by reset the base position before each respawn.

Fueredoriku committed:
Merge pull request #93 from hackerspace-ntnu/fix/respawn-bug

Fixed a bug where the player would spawn further and further away fro…

ThomasJGabrielsen committed:
Merge pull request #90 from hackerspace-ntnu/fix/unigolem-spawning

Fix unigolem spawning

ThomasJGabrielsen committed:
Adjusted instructions text

Fueredoriku committed:
Merge pull request #94 from hackerspace-ntnu/misc/buid-settings

Misc/buid settings

ddabble committed:
Merge pull request #91 from hackerspace-ntnu/feature/non-occupying-scenery

Add non-occupying scenery

Fueredoriku committed:
Refactored terrain on scene with new decor and removed Unigolem from scene.

ThomasJGabrielsen committed:
Merge pull request #95 from hackerspace-ntnu/scene/refactor

Scene update for demo

ThomasJGabrielsen reviewed (approved):
Looks good, I like how little has changed.

jesper2k reviewed (approved):
Works well 💯 👍

ddabble committed:
Fixed empty cell array in HexGrid + updated navmesh

Also clamped dealt damage in `HealthLogic.OnReceiveDamage()`, which was what was *actually* intended in bbe59129fab137720b860cab58ac665c0390063d.

ThomasJGabrielsen committed:
Merge pull request #97 from hackerspace-ntnu/fix/scene-and-navmesh

Fixed empty cell array in HexGrid + updated navmesh
